### PR TITLE
gitlab: add registry route53 registration

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- add the missing registry dns 

already applied in the environment

### CODE REVIEW

File: terraform/aws/modules/gitlab/gitlab-route53.tf [hunk 1]
```diff
@@ -8,14 +8,44 @@ data "kubernetes_service" "gitlab_nginx" {
...
 
+
+data "aws_elb" "gitlab_nginx_elb" {
+  # hostname have the name of the LB: internal-a6c7cc5f1622422ea8cb712fe8s21b48-995323897.us-east-1.elb.amazonaws.com need to get the name in order to ge tthe zone_id that will be used to create the A record with alias
+  name = split("-", data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname).1
+  depends_on = [
+    helm_release.gitlab,
+  ]
+}
+
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/gitlab/gitlab-route53.tf:19]: this one can be replaced by:
`data "aws_elb_hosted_zone_id" "main" {}`

File: terraform/aws/modules/gitlab/gitlab-route53.tf [hunk 2]
```diff
@@ -8,14 +8,44 @@ data "kubernetes_service" "gitlab_nginx" {
...
 
+
+resource "aws_route53_record" "registry_record" {
+  zone_id = var.private_hosted_zoneid
+  name    = "registry"
+  type    = "A"
+
+  alias {
+    name                   = data.kubernetes_service.gitlab_nginx.load_balancer_ingress.0.hostname
+    zone_id                = data.aws_elb.gitlab_nginx_elb.zone_id
```
Comment (angeloskyratzakos) [inline terraform/aws/modules/gitlab/gitlab-route53.tf:44]: and this one to be:
`zone_id                = data.aws_elb_hosted_zone_id.main.id`

File: terraform/aws/modules/gitlab/gitlab-route53.tf (hunk 1 above)
Comment (angeloskyratzakos) [reply terraform/aws/modules/gitlab/gitlab-route53.tf:19]: reference: https://www.terraform.io/docs/providers/aws/d/elb_hosted_zone_id.html

Comment (cpanato) [reply terraform/aws/modules/gitlab/gitlab-route53.tf:19]: the problem is this elb is not created by terraform, i tried and did not work, we need to fetch it

File: terraform/aws/modules/gitlab/gitlab-route53.tf (hunk 2 above)
Comment (cpanato) [reply terraform/aws/modules/gitlab/gitlab-route53.tf:44]: works without this

File: terraform/aws/modules/gitlab/gitlab-route53.tf (hunk 1 above)
Comment (angeloskyratzakos) [reply terraform/aws/modules/gitlab/gitlab-route53.tf:19]: I thought it didn't matter as the zone IDs are the same for the LBs in each region.
https://docs.aws.amazon.com/general/latest/gr/elb.html
Let me check it quickly

Comment (cpanato) [reply terraform/aws/modules/gitlab/gitlab-route53.tf:19]: it is already deploy as is and working :)

**Pull request merged**